tests/format_test.py

```diff
@@ -0,0 +1,190 @@
+        self.assertListEqual(prompt, ground_truth)
+
+        # wrong format
+        with self.assertRaises(TypeError):
+            model.format(*self.wrong_inputs)  # type: ignore[arg-type]
+
+    def test_ollama_chat(self) -> None:
+        """Unit test for the format function in ollama chat api wrapper."""
+        model = OllamaChatWrapper(
+            config_name="",
+            model_name="llama2",
+        )
+
+        # correct format
+        ground_truth = [
+            {"role": "system", "content": "You are a helpful assistant"},
+            {"role": "user", "content": "What is the weather today?"},
+            {"role": "assistant", "content": "It is sunny today"},
+        ]
+        prompt = model.format(*self.inputs)  # type: ignore[arg-type]
+        self.assertEqual(prompt, ground_truth)
+
+        # wrong format
+        with self.assertRaises(TypeError):
+            model.format(*self.wrong_inputs)  # type: ignore[arg-type]
+
+    def test_ollama_generation(self) -> None:
+        """Unit test for the generation function in ollama chat api wrapper."""
+        model = OllamaGenerationWrapper(
+            config_name="",
+            model_name="llama2",
+        )
+
+        # correct format
+        ground_truth = (
+            "system: You are a helpful assistant\nuser: What is "
+            "the weather today?\nassistant: It is sunny today"
+        )
+        prompt = model.format(*self.inputs)  # type: ignore[arg-type]
+        self.assertEqual(prompt, ground_truth)
+
+        # wrong format
+        with self.assertRaises(TypeError):
+            model.format(*self.wrong_inputs)  # type: ignore[arg-type]
+
+    @patch("google.generativeai.configure")
+    def test_gemini_chat(self, mock_configure: MagicMock) -> None:
+        """Unit test for the format function in gemini chat api wrapper."""
+        mock_configure.return_value = "client_dummy"
+
+        model = GeminiChatWrapper(
+            config_name="",
+            model_name="gemini-pro",
+            api_key="xxx",
+        )
+
+        # correct format
+        ground_truth = [
+            {
+                "role": "user",
+                "parts": [
+                    "system: You are a helpful assistant\nuser: What is the "
+                    "weather today?\nassistant: It is sunny today",
+                ],
+            },
+        ]
+
+        prompt = model.format(*self.inputs)  # type: ignore[arg-type]
+        self.assertListEqual(prompt, ground_truth)
+
+        # wrong format
+        with self.assertRaises(TypeError):
+            model.format(*self.wrong_inputs)  # type: ignore[arg-type]
+
+    def test_dashscope_chat(self) -> None:
+        """Unit test for the format function in dashscope chat api wrapper."""
+        model = DashScopeChatWrapper(
+            config_name="",
+            model_name="qwen-max",
+            api_key="xxx",
+        )
+
+        # correct format
+        # ground_truth = [
```

Comment:
Delete comments here or add a test case for advanced_format

Reply:
Modified accordingly